sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -764,3 +760,188 @@ def update_repository_properties(self, *args, **kwargs):
+            **kwargs
+        )
+        digest = response.http_response.headers['Docker-Content-Digest']
+        content = response.http_response.internal_response._content
+        content_stream = BytesIO(content)
+        
+        if not _validate_digest(content_stream, digest):
+            raise HttpResponseError("The requested digest does not match the digest of the received manifest.")
+        
+        manifest = _deserialize_manifest(content_stream)
+        return DownloadManifestResult(digest=digest, data=content_stream, manifest=manifest)
+
+    @distributed_trace
+    def download_blob(self, repository, digest, **kwargs):
+        # type: (str, str, **Any) -> DownloadBlobResult | None
+        """Download a blob that is part of an artifact.
+
+        :param str repository: Name of the repository
+        :param str digest: The digest of the blob to download.
+        :returns: DownloadBlobResult or None
+        :rtype: ~azure.containerregistry.DownloadBlobResult or None
+        :raises ValueError: If the parameter repository or digest is None.
+        """
+        if repository is None or digest is None:
+            raise ValueError("The parameter repository and digest cannot be None.")
+
+        result = self._client.container_registry_blob.get_blob(repository, digest, **kwargs)
+        
+        if not result:
+            return result
+        
+        data = BytesIO(result.response.internal_response.content)
+        digest = _compute_digest(data)
+        return DownloadBlobResult(data=data, digest=digest)
+            
+
+    @distributed_trace
+    def delete_manifest(self, repository, tag_or_digest, **kwargs):
+        # type: (str, str, **Any) -> None
+        """Delete a manifest. If the manifest cannot be found or a response status code of
+        404 is returned an error will not be raised.
+
+        :param str repository: Name of the repository the manifest belongs to
+        :param str tag_or_digest: Tag or digest of the manifest to be deleted
+        :returns: The Http response from the service.
+        :rtype: ~azure.core.pipeline.transport._requests_basic.RequestsTransportResponse
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        Example
+
+        .. code-block:: python
+
+            from azure.containerregistry import ContainerRegistryClient
+            from azure.identity import DefaultAzureCredential
+            endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+            client = ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="my_audience")
+            client.delete_manifest("my_repository", "my_tag_or_digest")
+        """
+        if _is_tag(tag_or_digest):
+            tag_or_digest = self._get_digest_from_tag(repository, tag_or_digest)
+
+        response = self._client.container_registry.delete_manifest(
+            repository, tag_or_digest, cls=_return_response, **kwargs
+        )
+        return response.http_response
```

Comment:
We are we returning the http response? This method shouldn't be returning anything.

Reply:
Why python does not prefer returning while it does return in other languages? E.g. [.Net](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs#L575)

Reply:
Have changed to return None.